deploy-service/common/src/main/java/com/pinterest/deployservice/scm/GithubManager.java

```diff
@@ -167,8 +168,8 @@ public String getUrlPrefix() {
     }
 
     @Override
-    public String getType() {
-        return TYPE;
+    public String getTypeName() {
+        return typeName;
     }
```

Comment:
Can this be implemented in the BaseManager?

Reply:
There's no way (to my knowledge) of getting the value of typeName into the class as an argument if a BaseManager object were to be instantiated, which is not possible since it is an abstract class

Reply:
I was thinking
```java
public class BaseManager {
    protected String typeName;
    public String getTypeName() {
        return typeName;
    }
} 

public class GirhubManager extends BaseManager {
      public GithubManager(String token, String appId, String appPrivateKeyKnox, String appOrganization, String 
        typeName, String apiPrefix, String urlPrefix) throws Exception {
            this.typeName = typeName;
  }
}
```

Reply:
Did you also remove the @Override implementation in GithubManager?

Reply:
Removing private field typeName in GithubManager did the trick. I will implement and commit the solution